db/db_impl/db_impl_write.cc

```diff
@@ -1541,8 +1541,11 @@ Status DBImpl::DelayWrite(uint64_t num_bytes,
   Status s;
   if (write_controller_.IsStopped()) {
     // If writes are still stopped, it means we bailed due to a background
-    // error
-    s = Status::Incomplete(error_handler_.GetBGError().ToString());
+    // error. Since the background error is now user visible and caused a
+    // write to fail, stop the DB and fail subsequent writes as well. There
+    // may be other writes in the queue and might cause inconsistency if the
+    // recovery succeeds and the queued writes are allowed to go through.
```

Comment:
It's true that if users use RocksDB in certain way, as what MyRocks does, any write error might not be recoverable. How about other use cases, where writes are more or less independent, so one write failure can be skipped or independently retried later?

Reply:
That's a good point. Should we introduce an option to control this behavior? And perhaps it could apply to other user write failures, such as IO error during WAL append. The automatic recovery will flush memtables and create a new WAL if there was a WAL append failure. Any writes during that time will be failed, but subsequent writes will succeed.

Reply:
I think if we want to do that, we probably should go with an option. Do we have an API that allows users to manually recover? If we do, the option can be for manual recovery only.

Reply:
Yes, ```DB::Resume()``` allows users to manually recover. I added an option to freeze the DB on a user write failure. I kept it independent of auto recovery, since the auto recovery can continue as long as its confined to the background and not visible to the user. If the freeze options is set and there is a user visible failure (either due to reason kWriteCallback or write controller stoppage), then we put the DB in read-only mode and cancel any ongoing recovery.